db/change_listener.go

```diff
@@ -59,6 +61,7 @@ func (listener *changeListener) Start(bucket base.Bucket, dbStats *expvar.Map) e
 		Backfill:   sgbucket.FeedNoBackfill,
 		Terminator: listener.terminator,
 		DoneChan:   make(chan struct{}),
+		GroupID:    listener.groupID,
```

Comment:
Same as above, regarding prefix storage vs. groupID.  DCP processing is particularly performance sensitive, definitely want to avoid the concatenation per DCP mutation if we can help it.

Reply:
I also don't think GroupID belongs in FeedArgs - it can just be a property on the listener and avoid the sgbucket change.

Reply:
Passing down the prefixes to `StartDCPFeed` would require a sgbucket change to modify the KVStore interface `StartDCPFeed` function to add the extra parameters. This would cause a lot of refactoring for other parts of the code that implement this (including to the walrus repo) that don't use the group ID and could cause unintended problems.

I think it would be best to use the FeedArguments (as the prefixes are being used internally for DCP) but instead of group ID like it is currently, to use the pre calculated prefixes (like suggested on https://github.com/couchbase/sg-bucket/pull/68). The prefixes would the SG Cfg prefix (used for DCP key filtering) and DCP checkpoint prefix.